### PR TITLE
add docker rest api agent

### PR DESCRIPTION
**Pull request by dexhunter**
docker agent for api-agent #121

Signed-off-by: Dixing (Dex) Xu <dixingxu@gmail.com>

### CODE REVIEW

File: src/agent/docker-rest-agent/server.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    res['data']['status'] = 'created'
+    res['data']['id'] = container.id
+    res['data']['public-grpc'] = '127.0.0.1:7050'
+    res['data']['public-raft'] = '127.0.0.1:7052'
+    return res
+
+@app.route('/api/v1/nodes/<id>', methods=['GET', 'POST'])
+def operate_node(id):
+
+    if request.method == 'POST':
```
Comment (yeasy) [inline src/agent/docker-rest-agent/server.py:53]: Why do we need this special if check?

File: src/agent/docker-rest-agent/server.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            res['data'] = sys.exc_info()[0]
+            res['msg'] = 'restart failed'
+            logging.debug(res)
+            raise
+        res['data']['status'] = 'restarted'
+    elif act == 'stop':
+        try:
+            container.stop()
+        except:
+            res['code'] = 'Fail'
```
Comment (yeasy) [inline src/agent/docker-rest-agent/server.py:83]: If you are using the same code snippet more than once, better wrap into some small method.

**Review by yeasy — COMMENTED** [review]
Besides, where are the testing cases?

File: src/agent/docker-rest-agent/server.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    'HLF_NODE_PEER_CONFIG':request.form.get('peer_config_file'),
+    'HLF_NODE_ORDERER_CONFIG':request.form.get('orderer_config_file'),
+    }
+
+
+    try:
+        # same as `docker run -dit yeasy/hyperledge-fabric:2.2.0 -e VARIABLES``
+        container = client.containers.run(request.form.get('img'), request.form.get('cmd'), detach=True, tty=True, stdin_open=True, name=request.form.get('name'), environment=env)
+    except:
+        res['code'] = 'Fail'
```
Comment (yeasy) [inline src/agent/docker-rest-agent/server.py:31]: Better use const to define the status code.

**Comment by dexhunter** [comment]
@yeasy I have addressed the comments and updated the code base, please help review. Thanks!

**Review by yeasy — APPROVED** [review]

**Pull request merged**